bftengine/src/bcstatetransfer/InMemoryDataStore.cpp

```diff
@@ -86,10 +86,13 @@ void InMemoryDataStore::setPrunedBlocksDigests(const std::vector<std::pair<Block
 }
 std::vector<std::pair<BlockId, Digest>> InMemoryDataStore::getPrunedBlocksDigests() { return prunedBlocksDigests; }
 
-void InMemoryDataStore::setCheckpointDesc(uint64_t checkpoint, const CheckpointDesc& desc) {
+void InMemoryDataStore::setCheckpointDesc(uint64_t checkpoint,
+                                          const CheckpointDesc& desc,
+                                          const bool checkIfAlreadyExists) {
   ConcordAssert(checkpoint == desc.checkpointNum);
-  ConcordAssert(descMap.count(checkpoint) == 0);
-
+  if (checkIfAlreadyExists) {
```

Comment:
nit: 
`concordAssertOR(!checkIfAlreadyExists, descMap.count(checkpoint) == 0);`
same for all other similar places.